src/Windows/Avalonia.Win32.Automation/Marshalling/SafeArray.cs

```diff
@@ -0,0 +1,320 @@
+﻿using System;
+using System.Buffers;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Avalonia.Controls.Documents;
+
+namespace Avalonia.Win32.Automation.Marshalling;
+
+#pragma warning disable CS0649 // Field is never assigned to, and will always have its default value
+#if NET7_0_OR_GREATER
+internal partial struct SafeArray
```

Comment:
Can we expose it as some opaque pointer type that has all the fields declared in a nested private struct?

It's not really safe to have a copy of SAFEARRAY struct even on the stack.

e. g.

```cs
internal struct SafeArray
{
     SAFEARRAY* _ptr;
     struct SAFEARRAY { fields here }
}
```

Reply:
Makes sense. Right now, it's only and only used as a pointer and only in source generated code. But since it's easy to make it safer, then sure.